### ISSUE TITLE
curve_fit broke

### ISSUE THREAD

**Issue by abieler** [JuliaOpt/Optim.jl#182]
Using Julia Version 0.4.3.
I used curve_fit for a while and after the last Pkg.update() it did not work anymore.

Seems commit cd297ad6b3f88d215b6556a82e466037ae91d70c from Feb. 16 is the cause.
(when checking out commits before that one everything works again.)

Here is the code and error I get when running the example from the LsqFit.jl

```julia
using LsqFit

# a two-parameter exponential model
# x: array of independent variables
# p: array of model parameters
model(x, p) = p[1]*exp(-x.*p[2])

# some example data
# xdata: independent variables
# ydata: dependent variable
xdata = linspace(0,10,20)
ydata = model(xdata, [1.0 2.0]) + 0.01*randn(length(xdata))

fit = curve_fit(model, xdata, ydata, [0.5, 0.5])
# fit is a composite type (LsqFitResult), with some interesting values:
#   fit.dof: degrees of freedom
#   fit.param: best fit parameters
#   fit.resid: residuals = vector of residuals
#   fit.jacobian: estimated Jacobian at solution

# We can estimate errors on the fit parameters,
# to get 95% confidence error bars:
errors = estimate_errors(fit, 0.95)

```

```
ERROR: DimensionMismatch("first dimension of A, 20, does not match length of y, 2")
 in gemv! at linalg/matmul.jl:225
 in A_mul_B! at linalg/matmul.jl:91
 in levenberg_marquardt at /home/abieler/.julia/v0.4/Optim/src/levenberg_marquardt.jl:88
 in lmfit at /home/abieler/.julia/v0.4/LsqFit/src/curve_fit.jl:29
 in curve_fit at /home/abieler/.julia/v0.4/LsqFit/src/curve_fit.jl:39

```

**Comment by pkofod** [comment]
This is the problem being addressed in #178 right? @KristofferC

**Comment by KristofferC** [comment]
Yes. It is just waiting for review.

**Comment by yuyichao** [comment]
Fixed by #178
